### DISCUSSION TITLE
Usage as a graph database

### DISCUSSION THREAD

**Discussion by vjpr** [tinyplex/tinybase#63] [Q&A]
Wondering how easy it is to use as a graph database.

Like say I need to recursively traverse a graph of relationships.

SQLite can be used as a graph database with recursive ctes for example.

**Comment by jamesgpearce** [comment]
Hm, I wouldn't say it's really optimal to do that. There is a way to mimic linked lists and relationships but as far as a generalized graph database goes, I would say no, sorry :(

**Reply by MentalGear** [reply]
Hey James, wondering if this ("not a good graph db", 3 years ago) is still the case, or whether there have been major changes that would support this? My use case is to support flexible, inter-shareable data for custom user-made applications, with key challenges including partial data availability (due to permissions), complex relationships, and robust reconnection and reconciliation after extended offline states (conflict resolution). Would much appreciate your input.
